### PR TITLE
W-10647630: fix issues with query string validations

### PR DESCRIPTION
**Pull request by juanchib**

### CODE REVIEW

File: src/main/java/org/mule/module/apikit/validation/attributes/ValidationUtils.java [hunk 1]
```diff
@@ -11,9 +11,10 @@
 public class ValidationUtils {
 
   static String escapeAndSurroundWithQuotesIfNeeded(Parameter facet, String value) {
-    return facet != null && (facet.isScalar() || (facet.isArray() && (!value.startsWith("{") && !value.startsWith("-"))))
-        ? facet.surroundWithQuotesIfNeeded(value.replace("\"", "\\\""))
-        : value;
+    return facet != null && value != null
+        && (facet.isScalar() || (facet.isArray() && (!value.startsWith("{") && !value.startsWith("-"))))
+            ? facet.surroundWithQuotesIfNeeded(value.replace("\"", "\\\""))
+            : value;
```
Comment (juanchib) [inline src/main/java/org/mule/module/apikit/validation/attributes/ValidationUtils.java:17]: This only fix the NPE issue

File: src/test/java/org/mule/module/apikit/validation/QueryStringRequestValidator.java [hunk 2]
```diff
@@ -36,9 +33,40 @@ public void validQueryString() throws MuleRestException {
   }
 
   @Test
-  public void invalidQueryString() throws MuleRestException {
-    expectedException.expect(InvalidQueryStringException.class);
-    expectedException.expectMessage("Invalid value for query string");
+  public void validUnionQueryString() throws MuleRestException {
+    MultiMap<String, String> queryParams = new MultiMap<>();
+    queryParams.put("stringOrNumber", "abc");
+    queryParams.put("nullableString", "123"); //TODO: deberia validar no valida por que on quotea para unions "abc" valida ok... feo
```
Comment (juanchib) [inline src/test/java/org/mule/module/apikit/validation/QueryStringRequestValidator.java:39]: here the problem/inconsistency with query params

File: src/test/java/org/mule/module/apikit/validation/QueryStringRequestValidator.java [hunk 3]
```diff
@@ -36,9 +35,61 @@ public void validQueryString() throws MuleRestException {
...
+    queryParams.put("lat", "12");
+    queryParams.put("long", "13");
+    queryParams.put("extraParam", "13");
+    testRestRequestValidatorBuilder
+        .withApiLocation("unit/query-string/api.raml")
+        .withMethod("GET")
+        .withRequestPath("/api/locations")
+        .withRelativePath("/locations")
+        .withQueryParams(queryParams)
+        .withQueryString("start=2&lat=12&long=13")
```
Comment (dcourtil) [inline src/test/java/org/mule/module/apikit/validation/QueryStringRequestValidator.java:50]: extraParam should be here despite the fact that it does not seem to have any impact

File: src/test/java/org/mule/module/apikit/validation/QueryStringRequestValidator.java [hunk 4]
```diff
@@ -36,9 +35,61 @@ public void validQueryString() throws MuleRestException {
...
+        .build()
+        .validateRequest();
+  }
+
+  @Test
+  public void notValidUnionQueryString() throws MuleRestException {
+    MultiMap<String, String> queryParams = new MultiMap<>();
+    queryParams.put("stringOrNumber", Collections.singleton(null));
+    queryParams.put("nullableString", Collections.singleton(null));
+    queryParams.put("nonNullableString", "13");
```
Comment (dcourtil) [inline src/test/java/org/mule/module/apikit/validation/QueryStringRequestValidator.java:79]: "123"

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java [hunk 5]
```diff
@@ -88,7 +88,7 @@ private static void validateQueryParametersStrictly(Map<String, Parameter> query
   }
 
   //only for raml 1.0
-  private static void validateQueryParamArray(String paramKey, Parameter expected, Collection<?> paramValues)
+  static void validateQueryParamArray(String paramKey, Parameter expected, Collection<?> paramValues)
```
Comment (JuanAller) [inline src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java:91]: why not private ?

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java [hunk 6]
```diff
@@ -102,8 +102,8 @@ private static void validateQueryParamArray(String paramKey, Parameter expected,
     validate(paramKey, expected, builder.toString());
   }
 
-  private static void validateQueryParam(MultiMap<String, String> queryParams, String paramKey, Parameter parameterDefinition,
-                                         String value)
+  static void validateQueryParam(MultiMap<String, String> queryParams, String paramKey, Parameter parameterDefinition,
```
Comment (JuanAller) [inline src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java:105]: idem

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryStringValidator.java [hunk 7]
```diff
@@ -94,36 +60,25 @@ private static String addDefaultValues(Map<String, Parameter> facetsWithDefault,
...
+  private static Map<String, Parameter> getFacetsWithDefaultValue(Map<String, Parameter> facets,
+                                                                  MultiMap<String, String> queryParams) {
     HashMap<String, Parameter> result = Maps.newHashMap();
     for (Entry<String, Parameter> entry : facets.entrySet()) {
       if (entry.getValue().getDefaultValue() != null) {
         result.put(entry.getKey(), entry.getValue());
       }
     }
+
+    for (Entry queryParam : queryParams.entrySet()) {
```
Comment (JuanAller) [inline src/main/java/org/mule/module/apikit/validation/attributes/QueryStringValidator.java:76]: Entry<String, String>

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java (hunk 5 above)
Comment (juanchib) [reply src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java:91]: it used to be reused by QueryStringValidator but yes now can be private

File: src/main/java/org/mule/module/apikit/validation/attributes/QueryStringValidator.java (hunk 7 above)
Comment (JuanAller) [reply src/main/java/org/mule/module/apikit/validation/attributes/QueryStringValidator.java:76]: we can do this in the previous for

**Pull request closed without merging**